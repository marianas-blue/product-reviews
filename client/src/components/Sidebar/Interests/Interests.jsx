/* eslint-disable react/button-has-type */
/* eslint-disable no-useless-constructor */
/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import { Descriptor, Border, Button, Title, Rating, Starbox, WrapButton } from './Interests.style';

class Interests extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Border>
        <Starbox>
          <Title>By consumer groups & interests</Title>
          <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/questionmark.png" alt="questionmark" height="20px" />
        </Starbox>
        <Descriptor>
          Headphones
          <Starbox>
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png" alt="stars" height="25px" />
            <Rating>4.2</Rating>
          </Starbox>
        </Descriptor>

        <Descriptor>
          Customers in New York
          <Starbox>
            <img src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png" alt="stars" height="25px" />
            <Rating>4.7</Rating>
          </Starbox>
        </Descriptor>
        <Descriptor style={{ textDecoration: 'none', color: '#7a7a7a' }}>
          Is this feature helpful?
          <WrapButton>
            <Button>Yes</Button>
            <Button>No</Button>
          </WrapButton>
        </Descriptor>
      </Border>
    );
  }
}

export default Interests;
