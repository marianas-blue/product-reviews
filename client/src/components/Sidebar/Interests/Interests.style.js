import styled from 'styled-components';

export const Descriptor = styled.div`
font-size: 14px;
display: flex;
justify-content: space-between;
align-items: center;
`;

export const Border = styled.div`
border-top: 1px solid #e7e7e7;
font-size .9em;
padding: 15px 15px 15px 0px
`;

export const Button = styled.button`
  background: #eff1f3;
  border-radius: 3px;
  border-color: #ADB1B8 #A2A6AC #8D9096;
  border-style: solid;
  border-width: 1px;
  cursor: pointer;
  display: inline-block;
  margin: 4px;
  text-align: center;
  vertical-align: middle;
  box-sizing: border-box;
  width: 50px;
  height: 20px;
  font-family: Lato,sans-serif;
  :hover {
    background: #e7e9ec;
  }
`;

export const Title = styled.button`
font-weight: bolder;
font-size: 17px;
line-height: 1.255;
border: none; 
padding: 0;
background-color: transparent;
outline: none;
text-align: left;
`;

export const Rating = styled.div`
  color: #797979;
  display: inline-block;
`;

export const Starbox = styled.div`
display: flex;
align-items: center;
margin-top: 7px;
margin-bottom: 7px;
`;

export const WrapButton = styled.div`
display: flex;
justify-content: space-between;
`;
