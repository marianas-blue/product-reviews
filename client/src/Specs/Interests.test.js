/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';

import Interests from '../components/Sidebar/Interests/Interests';

describe('Interests', () => {
  it('should render correctly', () => {
    const component = shallow(<Interests />);

    expect(component).toMatchSnapshot();
  });
});
