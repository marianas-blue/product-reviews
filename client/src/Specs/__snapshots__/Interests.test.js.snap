// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interests should render correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.button>
      By consumer groups & interests
    </styled.button>
    <img
      alt="questionmark"
      height="20px"
      src="https://s3.amazonaws.com/product-reviews-hr110/Icons/questionmark.png"
    />
  </styled.div>
  <styled.div>
    Headphones
    <styled.div>
      <img
        alt="stars"
        height="25px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png"
      />
      <styled.div>
        4.2
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div>
    Customers in New York
    <styled.div>
      <img
        alt="stars"
        height="25px"
        src="https://s3.amazonaws.com/product-reviews-hr110/Icons/stars.png"
      />
      <styled.div>
        4.7
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    style={
      Object {
        "color": "#7a7a7a",
        "textDecoration": "none",
      }
    }
  >
    Is this feature helpful?
    <styled.div>
      <styled.button>
        Yes
      </styled.button>
      <styled.button>
        No
      </styled.button>
    </styled.div>
  </styled.div>
</styled.div>
`;
